plugins/civchat2-paper/src/main/java/vg/civcraft/mc/civchat2/CivChat2Manager.java

```diff
@@ -345,6 +413,41 @@ public void addGroupChat(Player player, Group group) {
         scoreboardHUD.updateScoreboardHUD(player);
     }
 
+    
+    // Load banned words once when CivChat2Manager is created
+    private final Set<String> bannedWords;
+
+    private Set<String> loadBannedWords() {
+        
+        Set<String> words = new HashSet<>();
+        try {
+            File file = new File(instance.getDataFolder(), "banned-words.txt");
+            if (file.exists()) {
+                List<String> lines = Files.readAllLines(file.toPath());
+                for (String line : lines) {
+                    String cleanWord = line.trim().toLowerCase();
```

Comment:
```suggestion
                    String cleanWord = line.strip().toLowerCase();
```
Use #strip instead as that covers all whitespace characters while #trim only handles spaces.